src/v/cluster/self_test/netcheck.cc

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "cluster/self_test/netcheck.h"
+
+#include "cluster/logger.h"
+#include "random/generators.h"
+#include "utils/gate_guard.h"
+#include "vassert.h"
+#include "vlog.h"
+
+#include <seastar/core/coroutine.hh>
+
+#include <boost/range/irange.hpp>
+
+namespace {
+
+ss::future<cluster::netcheck_request>
+make_netcheck_request(model::node_id src, size_t sz) {
+    static const size_t fragment_size = 8192;
+    std::vector<size_t> fzs;
+    for (int64_t i = sz; i > 0; i -= fragment_size) {
+        fzs.push_back(i);
+    }
+    auto buffers = co_await ssx::parallel_transform(fzs, [](size_t size) {
+        ss::temporary_buffer<char> buf(size);
+        random_generators::fill_buffer_randomchars(buf.get_write(), size);
```

Comment:
the input parameter `sz` is divided up into chunks of `8192` pieces (fragments). `size` is each subsequent fragments size. This does no bounds checking (other then the limit of `size_t` input type itself) because bounds checking was already performed by `validate_options`. So `max` fragment size is 8192 bytes (it would take 15625 fragments to == 128MB) , which would comprise of 1 `iobuf` for 1 `netcheck_request`